context/app/static/js/components/detailPage/ProcessedData/ProcessedDataWorkspaceMenu.tsx

```diff
@@ -0,0 +1,131 @@
+
+import { useOpenDialog } from '../../workspaces/WorkspacesDropdownMenu/WorkspacesDropdownMenu';
+import { useCreateWorkspaceForm } from '../../workspaces/NewWorkspaceDialog/useCreateWorkspaceForm';
+import NewWorkspaceDialog from '../../workspaces/NewWorkspaceDialog/NewWorkspaceDialog';
+import AddDatasetsFromSearchDialog from '../../workspaces/AddDatasetsFromSearchDialog/AddDatasetsFromSearchDialog';
+import { useAppContext, useFlaskDataContext } from '../../Contexts';
+import { useTrackEntityPageEvent } from '../useTrackEntityPageEvent';
+
+interface ProcessedDataWorkspaceMenuProps {
+  button: React.ReactNode;
+}
+
+function ProcessedDataWorkspaceMenu({ button }: ProcessedDataWorkspaceMenuProps) {
+  const {
+    entity: { mapped_data_access_level, hubmap_id, uuid, entity_type, status },
+  } = useFlaskDataContext();
+  const isDataset = entity_type === 'Dataset';
+  const { isWorkspacesUser } = useAppContext();
+  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
+  const open = Boolean(anchorEl);
+  const track = useTrackEntityPageEvent();
+
+  const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    track({
+      action: 'Open Workspace Menu',
+      label: hubmap_id,
+    });
+    setAnchorEl(event.currentTarget);
+  };
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+
+  const {
+    control,
+    errors,
+    removeDatasets,
+    setDialogIsOpen: setOpenCreateWorkspace,
+    dialogIsOpen: createWorkspaceIsOpen,
+    ...rest
+  } = useCreateWorkspaceForm({
+    defaultName: hubmap_id,
+    initialSelectedDatasets: [uuid] ?? [],
+  });
+
+  const openEditWorkspaceDialog = useOpenDialog('ADD_DATASETS_FROM_SEARCH');
+
+  const trackCreateWorkspace = useEventCallback(() => {
+    track({
+      action: 'Start Creating Workspace',
+      label: hubmap_id,
+    });
+    setOpenCreateWorkspace(true);
+    handleClose();
+  });
+
+  const trackAddToWorkspace = useEventCallback(() => {
+    track({
+      action: 'Start Adding Dataset to Existing Workspace',
+      label: hubmap_id,
+    });
+    openEditWorkspaceDialog();
+    handleClose();
+  });
+
+  // Clone the button element and add the onClick handler
+  const buttonWithClickHandler = React.cloneElement(button as React.ReactElement, {
+    onClick: handleClick,
+    'aria-controls': open ? 'basic-menu' : undefined,
+    'aria-haspopup': 'true',
+    'aria-expanded': open ? 'true' : undefined,
+  });
+
+  const showWorkspaceButton =
+    mapped_data_access_level && hubmap_id && isDataset && isWorkspacesUser && status === 'Published';
+
+  if (!showWorkspaceButton) {
+    return null;
+  }
+
+  const options = [
+    {
+      children: 'Launch New Workspace',
+      onClick: trackCreateWorkspace,
+      icon: <WorkspacesIcon color="primary" fontSize="1.25rem" />,
+    },
+    {
+      children: 'Add to Workspace',
+      onClick: trackAddToWorkspace,
+      icon: <AddRounded color="primary" />,
```

Comment:
We don't need the `fontSize` prop here?

Reply:
Adding a `fontSize` prop of any value to this icon shrinks it, not sure why - but the default size matches designs.

Reply:
Icons directly imported from `@mui/material/icons` have a default fontSize of 1.25rem (medium): https://mui.com/material-ui/icons/#size

Icons we import from our icon map have their default font size set to 1rem by the default icon styles, so we end up needing to add the fontSize prop for those cases.